cmd/image/upload/upload.go

```diff
@@ -72,6 +82,31 @@ pvsadm image upload --bucket bucket1320 -f centos-8-latest.ova.gz -o centos8late
 			return err
 		}
 
+		if pkg.ImageCMDOptions.AccessKey != "" && pkg.ImageCMDOptions.SecretKey != "" {
+			s3Cli, err = client.NewS3Clientwithkeys(bxCli, opt.Region)
+			if err != nil {
+				fmt.Printf("err1: %+v\n", err)
+				return err
+			}
+
+			//Check if object exists or not
+			if opt.ObjectName == "" {
+				opt.ObjectName = filepath.Base(opt.ImageName)
+			}
```

Comment:
This can be moved outside in the beginning to avoid repeating in the block as well

Reply:
Done.